server/services/personService.test.ts

```diff
@@ -0,0 +1,75 @@
+import createError from 'http-errors'
+
+import PersonService from './personService'
+import { PrisonerClient } from '../data'
+import { prisonerFactory } from '../testutils/factories'
+import personUtils from '../utils/personUtils'
+
+jest.mock('../data/prisonerClient')
+
+describe('PersonService', () => {
+  const prisonerClient = new PrisonerClient('token') as jest.Mocked<PrisonerClient>
+  const prisonerClientBuilder = jest.fn()
+
+  const service = new PersonService(prisonerClientBuilder)
+
+  const token = 'token'
+
+  beforeEach(() => {
+    jest.resetAllMocks()
+    prisonerClientBuilder.mockReturnValue(prisonerClient)
+  })
+
+  describe('getPerson', () => {
+    describe('when the prisoner client finds the corresponding prisoner', () => {
+      it('converts the returned object to a Person', async () => {
+        const prisoner = prisonerFactory.build()
+
+        const person = personUtils.personFromPrisoner(prisoner)
+
+        prisonerClient.getPrisoner.mockResolvedValue(prisoner)
+
+        const result = await service.getPerson(token, prisoner.prisonerNumber)
+
+        expect(result).toEqual(person)
+
+        expect(prisonerClientBuilder).toHaveBeenCalledWith(token)
+        expect(prisonerClient.getPrisoner).toHaveBeenCalledWith(prisoner.prisonerNumber)
+      })
+    })
+
+    describe('when the prisoner client throws a 404 error', () => {
+      it('returns `null`', async () => {
+        const clientError = createError(404)
+        prisonerClient.getPrisoner.mockRejectedValue(clientError)
+
+        const notFoundPrisonNumber = 'NOT-FOUND'
+
+        const result = await service.getPerson(token, notFoundPrisonNumber)
+
+        expect(result).toEqual(null)
+
+        expect(prisonerClientBuilder).toHaveBeenCalledWith(token)
+        expect(prisonerClient.getPrisoner).toHaveBeenCalledWith(notFoundPrisonNumber)
+      })
+    })
+
+    describe('when the prisoner client throws any other error', () => {
+      it('re-throws the error', async () => {
+        const clientError = createError(500)
+        prisonerClient.getPrisoner.mockRejectedValue(clientError)
+
+        const prisonNumber = 'ABC123'
+
+        const expectedError = createError(500, {
+          userMessage: `Error fetching prisoner ${prisonNumber}.`,
+        })
```

Comment:
Might be worth checking if the second argument here makes a different to the test passing/failing - I found it didn't in a controller test

Reply:
Yeah this didn't work for me - I've gone with the approach you've taken on your PR for now. I'd like to figure out how to test this but don't want to sink too much time into this. Will add a card.